include/server.h

```diff
@@ -2,12 +2,20 @@
 #define SERVER_H
 
 #include <boost/asio.hpp>
-#include <vector>
+#include <map>
+#include <string>
 
 #include "connection.h"
-#include "server_info.h"
 #include "request_handler.h"
 
+struct ServerInfo {
+  int port;
+
+  /* Maps url path to its config */
+  std::map<std::string, RequestHandler*> path_to_handler;
```

Comment:
throughout our code base, we use multiple names to refer to the same thing:

* `path`
* `uri_prefix`

we should decide on one of the two in order to avoid confusion. My vote is for `uri_prefix` since it's part of the API (`RequestHandler::Init(const std::string& `**`uri_prefix`**`, const NginxConfig& config)`)